### PR TITLE
Check release is completely uploaded before restart

### LINKED ISSUE #91: Check if all the artifacts are updated and uploaded in nearup restart

### PR DESCRIPTION
**Pull request by nickmonad**
Resolves #91 

This ensures the new release intended to download is completely live on S3 before continuing the restart. Also wraps exceptions coming from `boto3` in a new name so we can catch them specifically and continue instead of stopping the watcher.

### CODE REVIEW

**Comment by nickmonad** [comment]
@chefsale I'm not sure if this is entirely the right approach, but should be a start.

File: nearuplib/util.py [hunk 1]
```diff
@@ -9,17 +10,65 @@
 from nearuplib.constants import S3_BUCKET
 
 
+class NetworkError(Exception):
```
Comment (chefsale) [inline nearuplib/util.py]: can you move this to a seperate file: nearuplib/exceptions.py

Comment (nickmonad) [reply nearuplib/util.py]: done!

**Review by chefsale — APPROVED** [review]
More more good stuff :1st_place_medal:

**Pull request merged**